commands/namespaces.go

```diff
@@ -0,0 +1,48 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openfaas/faas-cli/proxy"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	// Setup flags that are used by multiple commands (variables defined in faas.go)
+	namespacesCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	namespacesCmd.Flags().BoolVar(&tlsInsecure, "tls-no-verify", false, "Disable TLS validation")
+	namespacesCmd.Flags().StringVarP(&token, "token", "k", "", "Pass a JWT token to use instead of basic auth")
+
+	faasCmd.AddCommand(namespacesCmd)
+}
+
+var namespacesCmd = &cobra.Command{
+	Use:     `namespaces [--gateway GATEWAY_URL] [--tls-no-verify] [--token JWT_TOKEN]`,
+	Aliases: []string{"ns"},
+	Short:   "List OpenFaaS namespaces",
+	Long:    `Lists OpenFaaS namespaces either on a local or remote gateway`,
+	Example: `  faas-cli namespaces
+  faas-cli namespaces --gateway https://127.0.0.1:8080`,
+	RunE: runNamespaces,
+}
+
+func runNamespaces(cmd *cobra.Command, args []string) error {
+	gatewayAddress := getGatewayURL(gateway, defaultGateway, "", os.Getenv(openFaaSURLEnvironment))
+
+	namespaces, err := proxy.ListNamespacesToken(gatewayAddress, tlsInsecure, token)
+	if err != nil {
+		return err
+	}
+
+	printNamespaces(namespaces)
+
+	return nil
+}
+
+func printNamespaces(namespaces []string) {
+	fmt.Print("Namespaces:\n")
+	for _, v := range namespaces {
+		fmt.Printf("%s\n", v)
```

Comment:
Perhaps prefix with something? `- openfaas-fn`

Reply:
Yeah I will add that :+1:

Reply:
Done, updated the examples:
```
$ ./faas-cli namespaces --gateway=http://192.168.99.103:31112
Namespaces:
 - openfaas-fn
```